### DISCUSSION TITLE
Prisma on serverless - Any plan on addressing cold start issues?

### DISCUSSION THREAD

**Discussion by ShanonJackson** [prisma/prisma#5855] [Q&A]
We're currently using TypeORM and hate it, I love the Prisma approach to ORM. We're looking at swapping.

In some small internal tests we noticed that with prisma cold starts went from 600ms -> 1.6s.

Is there any plans to address this? I've noticed prisma comes in at like 3x the size of most other leading ORM's

**Comment by 2color** [comment]
Hey Shanon,

Which database are you using?

Here are some general performance tips when deploying Prisma to serverless platforms:
- Deploy the functions to the same region as the DB
- Instantiate Prisma Client outside the handler to reuse the connection to the database when the function is re-used
- Configure [provisioned concurrency](https://docs.aws.amazon.com/lambda/latest/dg/configuration-concurrency.html) on AWS Lambda – this allows you to keep a number of functions hot and ready to accept requests. 
- Use a connection pooler for the database, e.g. [PgBouncer](https://www.prisma.io/docs/concepts/components/prisma-client/deployment#usage-with-pgbouncer) to avoid high DB connection churn


Besides that, we're also working on some performance improvements. You can read more about the efforts [here](https://github.com/prisma/prisma/discussions/5819#discussioncomment-404842).

**Reply by ShanonJackson** [reply]
Thanks for the response!

We're not a fan of provisioned concurrency, but I know it works for many people; If cold starts are becoming an issue I feel provisioned concurrency is just a bandaid on a solution rather than addressing the root causes. We have deployed our function in the same region as our database. We have instantiated it outside the function, I've never heard of PgBouncer before so i'll check it out. 

This small benchmark actually came from a member of my team and not me myself; So its tough to know what the root cause I'll probably spend some time investigating myself.

I've read that thread (thanks for sending it through), and are confused about how calling "await connect" would be faster, when in both scenarios you have to wait for the connection. (wait before or wait on first lazy run).

if we start a $connect call but don't await it.... then everything on the main code path up until the first database query could gain some performance? or if $connect hasn't resolved before the first query will it still start a fresh connection? Hope that makes sense.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
